Applications/CMC/test/testCMCArm26_Millard.cpp

```diff
@@ -0,0 +1,115 @@
+ * Copyright (c) 2005-2016 Stanford University and the Authors                *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+// INCLUDE
+#include <OpenSim/Simulation/Model/Model.h>
+#include <OpenSim/Simulation/Model/AnalysisSet.h>
+#include <OpenSim/Tools/CMCTool.h>
+#include <OpenSim/Tools/ForwardTool.h>
+#include <OpenSim/Auxiliary/auxiliaryTestFunctions.h>
+
+#include <fstream>
+#include <thread>
+
+using namespace OpenSim;
+using namespace std;
+
+void testArm26(const std::string& setupFile, 
+               const std::string& resultFile) {
+    cout<<"\n******************************************************************" << endl;
+    cout << "*                             testArm26                          *" << endl;
+    cout << "******************************************************************\n" << endl;
+    CMCTool cmc(setupFile);
+    cmc.run();
+
+    Storage results(resultFile), temp("std_arm26_states.sto");
+    Storage *standard = new Storage();
+    cmc.getModel().formStateStorage(temp, *standard);
+
+    Array<double> rms_tols(0.02, 2*2+2*6); // activations within 2%, angles within .6 degrees
+    const string& muscleType = cmc.getModel().getMuscles()[0].getConcreteClassName();
+    string base = "testArm26 "+ muscleType;
+
+    if(muscleType != "Thelen2003Muscle"){
+        rms_tols[6] = 0.05;
+        rms_tols[8] = 0.05;
+        rms_tols[10] = 0.05;
+        rms_tols[12] = 0.05;
+        rms_tols[14] = 0.05;
+    }
+
+    CHECK_STORAGE_AGAINST_STANDARD(results, *standard, rms_tols, __FILE__, __LINE__, 
+        base+" failed");
+
+    
+    cout << "\n" << base <<" passed\n" << endl;
+}
+
+
+int main() {
+
+    SimTK::Array_<std::string> failures;
+
+    Object::renameType("Thelen2003Muscle", "Millard2012EquilibriumMuscle");
+
+    std::ifstream inFile{"arm26_Setup_CMC.xml"};
+    std::ofstream outFile{"arm26_Setup_CMC_Millard.xml"};
+    std::string line{};
+    while(std::getline(inFile, line)) {
+        // Find and replace muscle name.
+        {
+            std::string orig{"Thelen2003Muscle"};
+            std::string repl{"Millard2012EquilibriumMuscle"};
```

Comment:
`arm26_Setup_CMC.xml` doesn't use Thelen2003Muscle except in the <defaults> section, which doesn't actually matter. It is confusing to have this replacement here, I think, because it makes it seem like more is happening. Either remove or give a comment clarifying that this block doesn't actually change the muscle used in the model.